test/strings/search.jl

```diff
@@ -390,6 +390,34 @@ s_18109 = "fooα🐨βcd3"
     @test findall("aa", "aaaaaa", overlap=true) == [1:2, 2:3, 3:4, 4:5, 5:6]
 end
 
+# issue 37280
+@testset "UInt8, Int8 vector" begin
+    for VT in [Int8, UInt8]
+        A = VT[0x40, 0x52, 0x62, 0x52, 0x62]
+
+        @test findfirst(VT[0x30], A) === nothing
+        @test findfirst(VT[0x52], A) === 2:2
+        @test findlast(VT[0x30], A) === nothing
+        @test findlast(VT[0x52], A) === 4:4
+
+        pattern = VT[0x52, 0x62]
+
+        @test findfirst(pattern, A) === 2:3
+        @test findnext(pattern, A, 2) === 2:3
+        @test findnext(pattern, A, 3) === 4:5
+        @test findnext(pattern, A, 5) === nothing
+        @test findnext(pattern, A, 99) === nothing
+        @test_throws BoundsError findnext(pattern, A, -3)
```

Comment:
We should also check for `BoundsError` if the starting index is `> length(A) + 1`.   (*not* `> length(A)` … a starting index 1 past the end of the array seems to be allowed in `findnext`.)